include/exadg/incompressible_navier_stokes/spatial_discretization/generalized_newtonian_model.cpp

```diff
@@ -0,0 +1,334 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2023 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#include <exadg/incompressible_navier_stokes/spatial_discretization/generalized_newtonian_model.h>
+
+namespace ExaDG
+{
+namespace IncNS
+{
+template<int dim, typename Number>
+GeneralizedNewtonianModel<dim, Number>::GeneralizedNewtonianModel()
+  : Base(), generalized_newtonian_model(GeneralizedNewtonianViscosityModel::Undefined)
+{
+}
+
+template<int dim, typename Number>
+GeneralizedNewtonianModel<dim, Number>::~GeneralizedNewtonianModel()
+{
+}
+
+template<int dim, typename Number>
+void
+GeneralizedNewtonianModel<dim, Number>::initialize(
+  dealii::MatrixFree<dim, Number> const &                matrix_free_in,
+  std::shared_ptr<Operators::ViscousKernel<dim, Number>> viscous_kernel_in,
+  GeneralizedNewtonianModelData const &                  generalized_newtonian_model_data_in,
+  unsigned int                                           dof_index_velocity_in,
+  unsigned int                                           quad_index_velocity_linear_in)
+{
+  Base::initialize(matrix_free_in,
+                   viscous_kernel_in,
+                   dof_index_velocity_in,
+                   quad_index_velocity_linear_in);
+
+  generalized_newtonian_model_data_in.check();
+
+  generalized_newtonian_model = generalized_newtonian_model_data_in.generalized_newtonian_model;
+  viscosity_margin = static_cast<Number>(generalized_newtonian_model_data_in.viscosity_margin);
+  a                = static_cast<Number>(generalized_newtonian_model_data_in.a);
+  n                = static_cast<Number>(generalized_newtonian_model_data_in.n);
+  kappa            = static_cast<Number>(generalized_newtonian_model_data_in.kappa);
+  lambda           = static_cast<Number>(generalized_newtonian_model_data_in.lambda);
```

Comment:
Why do you extract the parameters here? To have a short name in the rest of the class, just use `data` for such a class member, because the context (generalized Newtonian model) is clear for the present class `GeneralizedNewtonianModel`.